mzLib/Proteomics/RetentionTimePrediction/Chronologer/ChronologerEstimator.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using TorchSharp;
+
+namespace Proteomics.RetentionTimePrediction.Chronologer
+{
+    public static class ChronologerEstimator
+    {
+        private static Chronologer ChronologerModel = new Chronologer();
+
+        /// <summary>
+        /// Uses the Chronologer model to predict C18 retention times (reported in % ACN).
+        /// Only modifications present in the Chronologer dictionary are supported.
+        /// Returns null if the sequence is not valid.
+        /// <code>
+        /// "Carbamidomethyl on C"
+        /// "Oxidation on M"
+        /// "Glu to PyroGlu"
+        /// "Phosphorylation on S"
+        /// "Phosphorylation on T"
+        /// "Phosphorylation on Y"
+        /// "Accetylation on K"
+        /// "Succinylation on K"
+        /// "Ubiquitination on K"
+        /// "Methylation on K"
+        /// "Dimethylation on K"
+        /// "Trimethylation on K"
+        /// "Methylation on R"
+        /// "Dimethylation on R"
+        /// </code>
+        /// </summary>
+        /// <param name="baseSequence"></param>
+        /// <param name="fullSequence"></param>
+        /// <returns></returns>
+        public static double? PredictRetentionTime(string baseSequence, string fullSequence)
+        {
+            var tensor = Tensorize(baseSequence, fullSequence);
+            if (tensor is null)
+                return null;
+
+            var prediction = ChronologerModel.Predict(tensor);
+            return prediction[0].ToDouble();
+        }
+
+        /// <summary>
+        /// Takes the base sequence and the full peptide sequence and returns a tensor for the Chronologer model.
+        /// The base sequence is the sequence without modifications and the full peptide sequence is the sequence with modifications.
+        /// The model is intended to be used with sequences of length 50 or less, and only supports modifications present in the Chronologer dictionary.
+        /// Sequences not supported by chronologer will return null.
+        /// </summary>
+        /// <param name="baseSequence"></param>
+        /// <param name="fullSequence"></param>
+        /// <returns></returns>
+        private static torch.Tensor Tensorize(string baseSequence, string fullSequence)
+        {
+            // Chronologer does not support metals
+            if (fullSequence.Contains("Metal") | (fullSequence.Contains("U") & !fullSequence.Contains("Unimod")))
```

Comment:
This should be || instead of |. || will short circuit (stop) if the left is false, but | will evaluate both regardless of the left being false or true.